ferrocene/ci/scripts/setup-qnx-toolchain.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: MIT OR Apache-2.0
+# SPDX-FileCopyrightText: The Ferrocene Developers
+
+# This script primarily exists to workaround the fact that on Windows BSD/GNU tar
+# are almost unusably slow. 
+
+import argparse
+import tarfile
+import boto3
+import zstandard
+
+TARBALL_BUCKET = "ferrocene-ci-mirrors"
+TARBALL_KEY = "manual/qnx/qnx710-deployment.tar.xz"
+
+def main():
```

Comment:
Next time I gotta do tarball stuff on Windows I';m gonna make this script more generic and reuse it. For now, I'd just like to get the QNX self-test working.